src/hnsw/external_index_socket.c

```diff
@@ -217,43 +278,45 @@ int create_external_index_session(const char                   *host,
     memcpy(init_buf, &hdr_msg, EXTERNAL_INDEX_MAGIC_MSG_SIZE);
     memcpy(init_buf + EXTERNAL_INDEX_MAGIC_MSG_SIZE, &index_params, sizeof(external_index_params_t));
     uint32 bytes_written
-        = send(client_fd, init_buf, sizeof(external_index_params_t) + EXTERNAL_INDEX_MAGIC_MSG_SIZE, 0);
+        = sendall(socket_con, init_buf, sizeof(external_index_params_t) + EXTERNAL_INDEX_MAGIC_MSG_SIZE, 0);
 
-    check_external_index_request_error(client_fd, bytes_written);
+    check_external_index_request_error(socket_con, bytes_written);
 
     if(params->pq) {
         external_index_send_codebook(
-            client_fd, buildstate->pq_codebook, params->dimensions, params->num_centroids, params->num_subvectors);
+            socket_con, buildstate->pq_codebook, params->dimensions, params->num_centroids, params->num_subvectors);
     }
 
-    uint32 buf_size = read(client_fd, &init_response, EXTERNAL_INDEX_INIT_BUFFER_SIZE);
+    uint32 buf_size = socket_con->read(socket_con, (char *)&init_response, EXTERNAL_INDEX_INIT_BUFFER_SIZE);
```

Comment:
buf_size type must must int32 since it might report -1 an later error handling function below compares it against -1